resources/server.rb

```diff
@@ -16,43 +16,131 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-property :server_string, String, name_property: true
-property :workgroup, String, default: 'SAMBA'
-property :interfaces, String, default: 'lo 127.0.0.1'
-property :hosts_allow, String, default: '127.0.0.0/8'
-property :bind_interfaces_only, String, default: 'no', equal_to: %w(yes no)
-property :load_printers, String, default: 'no', equal_to: %w(yes no)
-property :passdb_backend, String, default: 'tdbsam', equal_to: %w(ldapsam tdbsam smbpasswd)
-property :dns_proxy, String, default: 'no', equal_to: %w(yes no)
-property :security, String, default: 'user', equal_to: %w(user domain ADS share server)
-property :map_to_guest, String, default: 'Bad User'
-property :realm, String, default: ''
-property :password_server, String, default: '*'
-property :encrypt_passwords, String, default: 'yes', equal_to: %w(yes no)
-property :kerberos_method, String, default: 'secrets only', equal_to: ['secrets only', 'system keytab', 'dedicated keytab', 'secrets and keytab']
-property :log_level, String, default: '0'
-property :winbind_separator, String, default: '\\'
-property :idmap_config, String
-property :socket_options, String, default: '`TCP_NODELAY`'
-property :log_dir, String, default: lazy {
-  if platform_family?('rhel', 'fedora', 'amazon', 'suse')
-    '/var/log/samba/log.%m'
-  else
-    '/var/log/samba/%m.log'
-  end
-}
-property :max_log_size, [String, Integer], default: '5000' # 5M
-property :options, [Hash, nil], default: nil
-property :enable_users_search, [true, false], default: true
-property :shares, [Hash, nil], default: nil
-property :config_file, String, default: '/etc/samba/smb.conf'
-property :samba_services, Array, default: lazy {
-  if platform_family?('rhel', 'fedora', 'amazon', 'suse')
-    %w(smb nmb)
-  else
-    %w(smbd nmbd)
-  end
-}
+
+unified_mode true
+
+property :server_string,
+        String,
+        name_property: true,
+        description: 'Name of the server'
+
+property :workgroup,
+        String,
+        default: 'SAMBA',
+        description: 'The SMB workgroup to use'
+
+property :interfaces,
+        String,
+        default: 'lo 127.0.0.1',
+        description: 'Interfaces to listen on'
+
+property :hosts_allow,
+        String,
+        default: '127.0.0.0/8',
+        description: 'Allowed hosts/networks'
+
+property :bind_interfaces_only,
+        String,
```

Comment:
I think these boolean properties would be technically better accepting type `[true, false`] with a `coerce: proc { |p| p ? 'yes' : 'no' }` to get the string result for the config file.